app/scripts/modules/kubernetes/src/rawResource/component/group/RawResourceDetails.tsx

```diff
@@ -0,0 +1,232 @@
+import { KubernetesManifestService } from '../../../manifest/manifest.service';
+import { ManifestEvents } from '../../../pipelines/stages/deployManifest/manifestStatus/ManifestEvents';
+import { ManifestLabels } from '../../../manifest/ManifestLabels';
+import { IKubernetesManifestCondition, ManifestCondition } from '../../../manifest/status/ManifestCondition';
+import { RawResourceUtils } from '../RawResourceUtils';
+
+export interface IRawResourceDetailsProps {
+  app: any;
+  $stateParams: {
+    account: string;
+    application: string;
+    name: string;
+    region: string;
+  };
+}
+
+export interface IRawResourcesDetailState {
+  account: string;
+  application: string;
+  name: string;
+  region: string;
+  loading: boolean;
+  manifest?: IManifest;
+}
+
+Overridable('k8s.rawResource.details');
+export class RawResourceDetails extends React.Component<IRawResourceDetailsProps, IRawResourcesDetailState> {
+  constructor(props: IRawResourceDetailsProps) {
+    super(props);
+    this.state = {
+      account: null,
+      application: null,
+      name: null,
+      region: null,
+      loading: true,
+      manifest: null,
+    };
+  }
+  private destroy$ = new Subject();
+  private props$ = new Subject<IRawResourceDetailsProps>();
+  private unsubscribeManifest?: () => void;
+
+  public componentDidMount() {
+    this.setState({
+      account: this.props.$stateParams.account,
+      application: this.props.$stateParams.application,
+      name: this.props.$stateParams.name,
+      region: this.props.$stateParams.region,
+    });
+    const params = {
+      account: this.props.$stateParams.account,
+      location: this.props.$stateParams.region == '' ? '_' : this.props.$stateParams.region,
+      name: this.props.$stateParams.name,
+    };
+    this.unsubscribeManifest = KubernetesManifestService.subscribe(this.props.app, params, (manifest) => {
+      if (this.unsubscribeManifest != null) {
+        this.setState({ manifest, loading: false });
+      }
+    });
+  }
+
+  public componentWillReceiveProps(nextProps: IRawResourceDetailsProps) {
+    this.props$.next(nextProps);
+  }
+
+  public componentWillUnmount() {
+    this.destroy$.next();
+    if (this.unsubscribeManifest) {
+      this.unsubscribeManifest();
+      this.unsubscribeManifest = null;
+    }
+  }
+
+  private editRawResource() {
+    const { app } = this.props;
+    const { account, application, manifest } = this.state;
+
+    KubernetesManifestCommandBuilder.buildNewManifestCommand(
+      app,
+      manifest.manifest,
+      { app: application },
+      account,
+    ).then((builtCommand) => {
+      ManifestWizard.show({ title: 'Edit Manifest', application: app, command: builtCommand });
+    });
+  }
+
+  private deletedRawResource = () => {
+    const { app } = this.props;
+    const { account, manifest } = this.state;
+    const name = get(manifest, ['manifest', 'metadata', 'name'], null);
+    const kind = get(manifest, ['manifest', 'kind']);
```

Comment:
This typescript syntax sugar will retain type information.  I recommend using typescript optional chaining instead of lodash `.get()` in most scenarios for this reason
```suggestion
    const name = manifest?.manifest?.metadata?.name ?? null;
    const kind = manifest?.manifest?.kind;
```